src/client/hud.cpp

```diff
@@ -630,6 +630,15 @@ void drawItemStack(video::IVideoDriver *driver,
 		return;
 	}
 
+	// do not draw anything if most of the rect should be clipped
+	// TODO: find a way to clip properly
+	if (clip != nullptr) {
```

Comment:
You could use `if (ptr) {` everywhere. Not necessary to check against `nullptr` or `NULL`. Both are interpreted as `false`.

Reply:
Done.